matchms/Scores.py

```diff
@@ -121,6 +121,69 @@ def calculate_parallel(self) -> Scores:
         self._scores = self.similarity_function(self.references[:, 0], self.queries[0, :])
         return self
 
+    def scores_by_reference(self, references: ReferencesType) -> numpy.ndarray:
+        """Return all scores (not sorted) for the given reference spectrums.
+
+        Parameters
+        ----------
+        references
+            Single Spectrum or list or array of reference objects.
+        """
+        if not isinstance(references, (list, numpy.ndarray)):
+            references = [references]
+
+        selected_idx = []
+        for ref in references:
+            assert ref in self.references, "Given input not found in references."
+            selected_idx.append(int(numpy.where(self.references[:, 0] == ref)[0]))
+
+        return self._scores[selected_idx, :].copy()
```

Comment:
The function does not return to which query Spectrum each score belongs. Is this useful?

Reply:
I wasn't sure either.
Alternatives would be to return 
1. A list of tuples as you suggested in #78 : ``[(Spectrum, score)]`` where score can also be tuple of course.
2. A sliced ``Scores`` instance.

Reply:
(1) is of course much easier to implement, e.g. using ``return list(zip(self.queries[0], self._scores[selected_idx, :].copy()))``, for (2) we would have to create a new Scores instance and probably also add a slicing method or similar...

Reply:
List of tuples would be my preference. Getting another Scores object, would require another call to get for example nr matches.

Reply:
Agreed. I changed it.